src/api.spec.yaml

```diff
@@ -0,0 +1,595 @@
+          schema:
+            type: string
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ResourceClass"
+      responses:
+        "201":
+          description: Created a class in the resource pool
+          content:
+            "application/json":
+              schema: 
+                $ref: "#/components/schemas/ResourceClassWithId"
+        "404":
+          description: The resource pool does not exist
+          content:
+            "application/json":
+              schema: 
+                $ref: "#/components/schemas/ErrorResponse"
+      tags:
+        - classes
+    put:
+      summary: Update an existing resource class for a specific resource pool
+      parameters:
+        - in: path
+          name: resourcePoolId
+          required: true
+          schema:
+            type: string
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ResourceClassWithId"
+      responses:
+        "200":
+          description: Updated the classes in the resource pool
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/ResourceClassWithId"
+        "404":
+          description: The resource pool does not exist
+          content:
+            "application/json":
+              schema: 
+                $ref: "#/components/schemas/ErrorResponse"
+      tags:
+        - classes
+  /resourcePools/{resourcePoolId}/classes/{classId}:
+    get:
+      summary: Get a specific resource class in a specific resource pool
+      parameters:
+        - in: path
+          name: resourcePoolId
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: classId
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          description: The resource class that was requested
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/ResourceClassWithId"
+        "404": 
+          description: The resource class does not exist
+          content:
+            "application/json":
+              schema: 
+                $ref: "#/components/schemas/ErrorResponse"
+      tags:
+        - classes
+    delete:
+      summary: Remove the specific resource class
+      parameters:
+        - in: path
+          name: resourcePoolId
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: classId
+          required: true
+          schema:
+            type: string
+      responses:
+        "204":
+          description: The resource class was removed or did not exist in the first place
+      tags:
+        - classes
+  /resourcePools/{resourcePoolId}/users:
```

Comment:
I think we should have a separate `/users` endpoint and then here only have `GET` (list of users), `POST` (add a new user id) and `DELETE` (remove a user id), not put (editing an id mapping makes no sense).

Then we can also have `GET /users/{id}/resourcePools` to get all pools for a user. 
That feels cleaner and more in the spirit of REST to me.